src/CoreWCF.Primitives/src/CoreWCF/Security/MessageSecurityOverHttp.cs

```diff
@@ -0,0 +1,437 @@
+                //            break;
+                //        case MessageCredentialType.Windows:
+                //            oneShotSecurity = SecurityBindingElement.CreateKerberosBindingElement();
+                //            isKerberosSelected = true;
+                //            break;
+                //        case MessageCredentialType.IssuedToken:
+                //            oneShotSecurity = SecurityBindingElement.CreateIssuedTokenForCertificateBindingElement(IssuedSecurityTokenParameters.CreateInfoCardParameters(new SecurityStandardsManager(new WSSecurityTokenSerializer(emitBspAttributes)), this.algorithmSuite));
+                //            break;
+                //        default:
+                //            Fx.Assert("unknown ClientCredentialType");
+                //            throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new NotSupportedException());
+                //    }
+                //}
+                if (this.IsSecureConversationEnabled())
+                {
+                    result = SecurityBindingElement.CreateSecureConversationBindingElement(oneShotSecurity, true);
+                }
+                else
+                {
+                    result = oneShotSecurity;
+                }
+            }
+
+            // set the algorithm suite and issued token params if required
+            if (wasAlgorithmSuiteSet || (!isKerberosSelected))
+            {
+                result.DefaultAlgorithmSuite = oneShotSecurity.DefaultAlgorithmSuite = this.AlgorithmSuite;
+            }
+            else if (isKerberosSelected)
+            {
+                result.DefaultAlgorithmSuite = oneShotSecurity.DefaultAlgorithmSuite = SecurityAlgorithmSuite.KerberosDefault;
+            }
+
+            result.IncludeTimestamp = true;
+            oneShotSecurity.MessageSecurityVersion = version;
+            result.MessageSecurityVersion = version;
+            if (!isReliableSession)
+            {
+                result.LocalServiceSettings.ReconnectTransportOnFailure = false;
+               // result.LocalClientSettings.ReconnectTransportOnFailure = false;
+            }
+            else
+            {
+                result.LocalServiceSettings.ReconnectTransportOnFailure = true;
+               // result.LocalClientSettings.ReconnectTransportOnFailure = true;
+            }
+
+            if (this.IsSecureConversationEnabled())
+            {
+                string defaultServerIssuedTransitionTokenLifetimeString = "00:15:00";
+                TimeSpan defaultServerIssuedTransitionTokenLifetime = TimeSpan.Parse(defaultServerIssuedTransitionTokenLifetimeString, CultureInfo.InvariantCulture);
+                oneShotSecurity.LocalServiceSettings.IssuedCookieLifetime = defaultServerIssuedTransitionTokenLifetime;
+                //TODO SpNego when port, remove above and enable below.
```

Comment:
Create a static readonly in this class which has the value TimeSpan.FromMinutes(15) called defaultServerIssuedTransitionTokenLifetime and then assign it. The SpnegoTokenAuthenticator static was just used because it was convenient and a string was used because of tying into config.